python/pyarrow/tests/test_compute.py

```diff
@@ -1092,11 +1092,8 @@ def test_strptime():
 def test_count():
     arr = pa.array([1, 2, 3, None, None])
     assert pc.count(arr).as_py() == 3
-    assert pc.count(arr, count_mode='count_non_null').as_py() == 3
-    assert pc.count(arr, count_mode='count_null').as_py() == 2
-
-    with pytest.raises(ValueError, match="'zzz' is not a valid count_mode"):
-        pc.count(arr, count_mode='zzz')
```

Comment:
Why remove this?

Reply:
It didn't make sense to keep as it's not a mode but rather a boolean variable now and string would evaluate to true.
I reintroduced a test for the integer (min_count).